src/model/keychain.js

```diff
@@ -136,3 +143,86 @@ export async function hasKey(key) {
   }
   return null;
 }
+
+export async function restoreBackupIntoKeychain(backedUpData) {
+  // Access control config per each type of key
+  const publicAccessControlOptions = {
+    accessible: ACCESSIBLE.ALWAYS_THIS_DEVICE_ONLY,
+  };
+
+  let privateAccessControlOptions = {};
+  const canAuthenticate = await canImplyAuthentication({
+    authenticationType: AUTHENTICATION_TYPE.DEVICE_PASSCODE_OR_BIOMETRICS,
+  });
+
+  let isSimulator = false;
+
+  if (canAuthenticate) {
+    isSimulator = __DEV__ && (await DeviceInfo.isEmulator());
+  }
+  if (canAuthenticate && !isSimulator) {
+    privateAccessControlOptions = {
+      accessControl: ACCESS_CONTROL.USER_PRESENCE,
+      accessible: ACCESSIBLE.WHEN_UNLOCKED_THIS_DEVICE_ONLY,
+    };
+  }
+
+  Object.keys(backedUpData).forEach(async key => {
+    const value = backedUpData[key];
+    let accessControl = publicAccessControlOptions;
+    if (
+      (key.indexOf('rainbowSeedPhrase') !== -1 ||
+        key.indexOf('rainbowPrivateKey') !== -1) &&
+      key !== 'rainbowSeedPhraseMigratedKey'
+    ) {
+      accessControl = privateAccessControlOptions;
+    }
+    if (typeof value === 'string') {
+      await saveString(key, value, accessControl);
+    } else {
+      await saveObject(key, value, accessControl);
+    }
+  });
+
+  // Save the migration flag
+  // to prevent this flow in the future
+  await saveString(
```

Comment:
Hm this is a note for now: do we actually need to save this here?

Reply:
We do

Reply:
From what we discussed: possible a user has the old seed phrase key and has not migrated yet, so we would be taking their secrets as is. We probably want to migrate at the time of backing up, and so we won't need to set true here

Reply:
@brunobar79 I'm thinking this is the most urgent thing for merging, others I think we can do after no problem